### PR TITLE
Add get user info API

### PR DESCRIPTION
**Pull request by UtopiaBeam**
- add `JwtMiddleware`
- add `AuthGuard`
- add `/user/me` endpoint
- remove `.dockerignore`
- add Bearer authorization in swagger

### CODE REVIEW

**Comment by beesk135** [comment]
why do u have to remove dockerignore?

**Comment by UtopiaBeam** [comment]
> why do u have to remove `.dockerignore`?

We do not use it right now, and it seems not recognized by docker-compose.

File: src/guards/auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        @Inject('UserService') private readonly userService: UserService,
+    ) {}
+
+    canActivate(
+        context: ExecutionContext,
+    ): boolean | Promise<boolean> | Observable<boolean> {
+        const req: Request = context.switchToHttp().getRequest();
+        if (req.userId) {
+            return this.userService.exists(req.userId);
+        }
```
Comment (beesk135) [inline src/guards/auth.guard.ts]: request `userId` is a parameterr in Request?

File: src/guards/auth.guard.spec.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { AuthGuard } from './auth.guard';
+import { UserService } from '../user/user.service';
+import { Test } from '@nestjs/testing';
+
+describe('AuthGuard', () => {
+    let userService: UserService;
```
Comment (beesk135) [inline src/guards/auth.guard.spec.ts]: what is the purpose of function `describe` ?

**Comment by beesk135** [comment]
well written code style

**Pull request merged**